program-runtime/src/invoke_context.rs

```diff
@@ -170,7 +217,15 @@ impl<'a> InvokeContext<'a> {
         feature_set: Arc<FeatureSet>,
         blockhash: Hash,
         lamports_per_signature: u64,
+        current_accounts_data_len: u64,
     ) -> Self {
+        let accounts_data_meter = {
+            assert!(current_accounts_data_len <= MAX_ACCOUNTS_DATA_LEN);
+            AccountsDataMeter::new_ref(
+                // SAFETY: The assert above guarantees the subtraction operation will not wrap
+                MAX_ACCOUNTS_DATA_LEN.wrapping_sub(current_accounts_data_len),
```

Comment:
Why not saturating sub?

Reply:
Since we assert that current accounts data len is less than the max, we know there won't be any overflow. So no need to pay for the checks in the `saturating_sub()`.

Reply:
Optionally I could remove the assert and change this to a saturating_sub too.

Reply:
Changed this to saturating_sub() since I've changed the assert to a debug_assert.